gradoop-hbase/src/main/java/org/gradoop/flink/io/impl/hbase/HBaseDataSink.java

```diff
@@ -60,6 +60,15 @@
 public class HBaseDataSink extends HBaseBase<GraphHead, Vertex, Edge>
   implements DataSink {
 
+  /**
+   * Temporal output path for mapreduce step
+   */
+  private static final String MAPRED_OUTPUT_PATH = "/tmp/mapred-out/";
```

Comment:
I don't see why this needs to be hardcoded. Typically, you would set those options in the corresponding config file, `mapred-site.xml` in that case. Also the property is deprecated: https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-common/DeprecatedProperties.html

I'm fine with having a OS-independent default param  (which that one is not) which is set if it's not set from the outside. However, now the user-defined one is always overwritten.

Reply:
I know it's deprecated, but in the current implementation (see [HadoopOutputFormatBase](https://github.com/apache/flink/blob/master/flink-connectors/flink-hadoop-compatibility/src/main/java/org/apache/flink/api/java/hadoop/mapreduce/HadoopOutputFormatBase.java) line 175) uses this property without checking if its present. Therefore in an example application and a UnitTest of the flink-connectors module, this property is set hardcoded. (See the ticket #760 for the links to the code)

OK, I'll try to check if this is set in the xml and if not, we set a OS-independent default value. Do you have a suggestion for a path?

Reply:
I'm not argueing that it should not be set. I'm just saying it does not need to be set inside the `HBaseDataSink` (where the user has no access to). In the Flink code, they set in within the Example (which is user code). https://github.com/apache/flink/blob/master/flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseWriteExample.java#L79 Maybe provide as a parameter to the HBaseDataSink? In a test, you can use https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html for creating a temp folder.

Reply:
I noticed that a specific mapred-site.xml can be specified in the HBaseConfiguration before giving it to the HBaseEPGMStore object.
```
Configuration hBaseConfiguration = HBaseConfiguration.create();
hBaseConfiguration.addResource("mapred-site.xml");
```
With this method, the user can specify the necessary property. Thus, no change is needed and I'll close the PR and the ticket.